### PR TITLE
Fix for Issue #83

### PR DESCRIPTION
**Pull request by dalehenrich**
Per #83, added code to gemstone/run.sh to cache extents and gsDevKitCommandLine image (and Pharo3.0 download) when dependency cache is enabled in `.travis.yml`:

```yml
cache:
  directories:
    - $TRAVIS_BUILD_DIR/_cache
```
When running local script the env var `GS_TRAVIS_CACHE_ENABLED` can be used to control caching behavior - the default is to use the cache, so the following will turn off the cache:

```
export GS_TRAVIS_CACHE_ENABLED=false
```

The above code has been included in the `.travis.yml` file for smalltalkCI (commented out), if dependency caching is enabled gemstone uses 126MB zipped (includes a Pharo-3.0 image. changes and vm). The squeak and pharo images and vms add another 200MB (314MB cache size according to travis for full build), but I think there's some cruft in there as the squeak builds leave tgz files for images and vm in the `_cache` directory. If that is not an acceptable cache size then we may want to leave the code commented out (or removed) ... but if the cache isn't too large the GemStone linux builds are about 3 times faster ....

I am leaving  the expected failures for the GemStone OSX builds enabled. The api rate limit is no longer an issue, but I am still seeing occasional network errors causing build failures on the OSX builds for GemStone,restarting the build can get past those issues, unlike the api rate limit which seemd to be perpetually exceeded on the OSX boxes -- but it is annoying to get these random failures.

### CODE REVIEW

**Comment by dalehenrich** [comment]
I've got to head out, but it looks like there might be a logic error in the gemstone/run.sh script ... I will try to check later tonight ...

**Comment by fniephaus** [comment]
Thanks, I will try to review this tomorrow.

**Comment by dalehenrich** [comment]
Yep, I had a typo that caused env vars controlling caching to be set incorrectly - `gemstone/run.sh` was updating the `_cache` dir without dependency caching actually being enabled - fixed typo and verified the logic was correct ...

**Comment by dalehenrich** [comment]
[build failure](https://travis-ci.org/hpi-swa/smalltalkCI/jobs/116809819#L1545) is a network error --- restart and it should pass

File: gemstone/run.sh [hunk 1]
```diff
@@ -109,6 +179,18 @@ run_build() {
       ;;
   esac
 
+  # To bypass cached behavior for local build, export GS_TRAVIS_CACHE_ENABLED
+  #   before calling run.sh
+  if [ "${GS_TRAVIS_CACHE_ENABLED}x" = "x" ]; then
```
Comment (fniephaus) [inline gemstone/run.sh]: Why is this necessary? One can easily delete the cache by removing the `_cache` directory.
I implemented caching for Squeak and Pharo, because I didn't want to download everything for every build. So that is very useful when I run builds locally to debug a problem.

Comment (dalehenrich) [inline gemstone/run.sh]: My main thought was that I wanted to be able to test the code locally when `GS_TRAVIS_CACHE_ENABLED=false`, since that is a separate code path in gemstone/run.sh.

Comment (fniephaus) [inline gemstone/run.sh]: I'm happy to leave it in if it helps :)
